gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -874,6 +890,9 @@
    * Hive registration properties
    */
   public static final String HIVE_REGISTRATION_POLICY = "hive.registration.policy";
+  public static final String HIVE_REG_PUBLISHER_CLASS = "hive.reg.publisher.class";
```

Comment:
I actually believe we should stop adding configurations into this `ConfigurationKey` class but put all these configurations into most relevant class. The reason is if you put everything here and make them public static final, any constructs within Gobblin that carrying state can modify it without any restriction, and also you lost context on what does this configuration really mean.

Reply:
agreed